mkdocs/nav.py

```diff
@@ -186,6 +187,18 @@ def set_canonical_url(self, base):
             base += '/'
         self.canonical_url = utils.urljoin(base, self.abs_url.lstrip('/'))
 
+    def set_edit_url(self, repo_url, edit_uri):
+        if not repo_url.endswith('/'):
+            repo_url += '/'
+        if not edit_uri:
+            self.edit_url = repo_url
+        else:
+            if not edit_uri.endswith('/'):
+                edit_uri += '/'
+            self.edit_url = utils.urljoin(
+                repo_url+edit_uri,
+                self.input_path.lstrip('/'))
+
```

Comment:
Just took a closer look as this method and I have  few concerns:

1. You are using `utils.urljoin` to join the parts, so you shouldn't need to also be adding `/` manually.

2. I don't see were the page specific path is being added here. Shouldn't `self.input_path` be appended to the end of each url (see this [comment][1])?

[1]: https://github.com/mkdocs/mkdocs/pull/752#issuecomment-173363860